test/Sentry.Unity.Android.Tests/SentryNativeAndroidTests.cs

```diff
@@ -81,28 +92,68 @@ public void Configure_IL2CPP_ReInitializesNativeBackend(bool il2cpp, bool expect
         _sentryUnityInfo.IL2CPP = il2cpp;
         Assert.False(_reinstallCalled); // Sanity check
 
-        SentryNativeAndroid.Configure(new(), _sentryUnityInfo);
+        SentryNativeAndroid.Configure(_options, _sentryUnityInfo);
 
         Assert.AreEqual(expectedReinstall, _reinstallCalled);
     }
 
     [Test]
     public void Configure_NativeAndroidSupportDisabled_DoesNotReInitializeNativeBackend()
     {
-        var options = new SentryUnityOptions { AndroidNativeSupportEnabled = false };
-        SentryNativeAndroid.Configure(options, _sentryUnityInfo);
+        _options.AndroidNativeSupportEnabled = false;
+        SentryNativeAndroid.Configure(_options, _sentryUnityInfo);
         Assert.False(_reinstallCalled);
     }
 
     [Test]
     public void Configure_NoInstallationIdReturned_SetsNewDefaultUserId()
     {
-        var options = new SentryUnityOptions();
-        var sentryJava = SentryNativeAndroid.SentryJava as TestSentryJava;
-        Assert.NotNull(sentryJava);
-        sentryJava!.InstallationId = string.Empty;
+        _testSentryJava.InstallationId = string.Empty;
 
-        SentryNativeAndroid.Configure(options, _sentryUnityInfo);
-        Assert.False(string.IsNullOrEmpty(options.DefaultUserId));
+        SentryNativeAndroid.Configure(_options, _sentryUnityInfo);
+        Assert.False(string.IsNullOrEmpty(_options.DefaultUserId));
+    }
+
+    [Test]
+    public void Configure_DefaultConfigurationSentryJavaNotPresent_LogsErrorAndReturns()
+    {
+        _testSentryJava.SentryPresent = false;
+
+        SentryNativeAndroid.Configure(_options, _sentryUnityInfo);
+
+        Assert.IsTrue(_logger.Logs.Any(log =>
+            log.logLevel == SentryLevel.Error &&
+            log.message.Contains("Sentry Java SDK is missing.")));
+
+        Assert.Null(_options.ScopeObserver);
+    }
+
+    [Test]
+    public void Configure_NativeAlreadyInitialized_LogsAndConfigures()
+    {
+        _testSentryJava.Enabled = true;
+
+        SentryNativeAndroid.Configure(_options, _sentryUnityInfo);
+
+        Assert.IsTrue(_logger.Logs.Any(log =>
+            log.logLevel == SentryLevel.Debug &&
+            log.message.Contains("The native SDK is already initialized")));
+
+        Assert.NotNull(_options.ScopeObserver);
+    }
+
+    [Test]
+    public void Configure_NativeInitFails_LogsErrorAndReturns()
```

Comment:
There was so many changes done above. Are these 3 tests covering all of those?

Some values should be default false, etc.

Reply:
The only behavioral change that we did was introduce `Init` to the `Configure`. The only real thing to check is whether `Init` gets called and validate the fallbacks.

The rest is getting covered by our integration tests. I'll fix those up next as part of https://github.com/getsentry/sentry-unity/pull/1865